src/Features/CSharp/Portable/MakeMethodAsynchronous/CSharpMakeMethodAsynchronousCodeFixProvider.cs

```diff
@@ -119,11 +119,10 @@ TypeSyntax MakeGenericType(string type, ITypeSymbol typeArgumentFrom)
 
         private static bool IsIterator(IMethodSymbol x)
         {
-            return x.Locations
-                .Any(l => ContainsYield(l));
+            return x.Locations.Any(l => ContainsYield(l.FindNode(cancellationToken: default)));
 
-            bool ContainsYield(Location l)
-                => l.FindNode(default).DescendantNodes().Any(n => IsYield(n));
+            bool ContainsYield(SyntaxNode node)
+                => node.DescendantNodes(n => n == node || !n.IsKind(SyntaxKind.LocalFunctionStatement)).Any(n => IsYield(n));
```

Comment:
Very tiny nit: consider just using `.IsReturnableConstruct` instead of `IsKind`. I understand that will have no effect on all correct code because lambdas cannot have yields but it seems appropriate to stop the search at any kind of function-like thing, and that this method shouldn't really have to think about which of those can contain yields and which cannot.

Reply:
Or alternatively, feel free to tell me to shut up at this point :smile:

Reply:
Yes, I caught that case with a test. I'd missed it initially too ;-)